chore: Notifications Settings Polishing

Jonathansoufer opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR are part of a series of PRs to implement the new notifications feature on MetaMask. This brings UI elements together to present our users:

- Notification SETTINGS Screen

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this Settings screen, open the new Notifications option and enable it.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

https://github.com/MetaMask/metamask-mobile/assets/44679989/80029408-ea8d-424e-9f1d-735423d5ff02

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

Commit:
chore: Polishing of Notification's settings

Commit:
chore: add Engine services states from notifications controllers

Commit:
chore: add translations

Commit:
chore: updates snapshots

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Commit:
chore: add snaps fence

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Cal-L commented on app/components/Views/Settings/NotificationsSettings/index.tsx:
```diff
@@ -15,104 +17,98 @@ import { useAccounts } from '../../../../components/hooks/useAccounts';
+  const isFullScreenModal = route?.params?.isFullScreenModal;
+  // Style
+  const { colors } = theme;
+  const styles = createStyles(colors);
```
use `useStyles` instead, which will automatically expose theme in the style sheet function

Jonathansoufer replied:
adjusted

Cal-L commented on app/components/Views/Settings/NotificationsSettings/index.tsx:
```diff
@@ -15,104 +17,98 @@ import { useAccounts } from '../../../../components/hooks/useAccounts';
+  useEnableNotifications,
+} from '../../../../util/notifications/hooks/useNotifications';
+import { CONSENSYS_PRIVACY_POLICY } from '../../../../constants/urls';
+import { useAccountSettingsProps } from '../../../../util/notifications/hooks/useSwitchNotifications';
```
I may have missed it but where is the file `useSwitchNotifications` in this PR?

Jonathansoufer replied:
It was introduced [here](https://github.com/MetaMask/metamask-mobile/issues/10340) as part of the break down effort PRs.

Cal-L commented on locales/languages/en.json:
```diff
@@ -1979,7 +1979,7 @@
     "sent": "Sent to {{address}}",
     "menu_item_title": {
-      "metamask_swap_completed": "Swapped {{symbol1}} for {{symbol2}}",
+      "metamask_swap_completed": "Swapped {{symbolIn}} for {{symbolOut}}",
```
Where is the change in the PR that utilizes this change?

Jonathansoufer replied:
[Here](https://github.com/MetaMask/metamask-mobile/blob/5a892bbfa91a0850c39f0a9225a993598eaef1a3/app/util/notifications/notification-states/swap-completed/swap-completed.tsx)

Cal-L commented on locales/languages/en.json:
```diff
@@ -2006,7 +2006,7 @@
       "title_stake": "Staked {{symbol}}",
       "title_unstake_completed": "Unstaking completed",
-      "title_swapped": "Swapped {{symbol1}} to {{symbol2}}",
+      "title_swapped": "Swapped {{symbolIn}} to {{symbolOut}}",
```
Where is the change in the PR that utilizes this change?

Jonathansoufer replied:
[Here](https://github.com/MetaMask/metamask-mobile/blob/5a892bbfa91a0850c39f0a9225a993598eaef1a3/app/util/notifications/notification-states/swap-completed/swap-completed.tsx)

Cal-L reviewed (commented):
Left some comments

Jonathansoufer commented on app/components/Views/Settings/NotificationsSettings/sectionHeader.tsx:
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable import/prefer-default-export */
```
Adjusted.

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Commit:
chore: adjust style and export

Commit:
chore: add missing keys

Commit:
chore: refactor controllers import

Commit:
chore: refactore Engine usage

Commit:
chore: add mock state

Commit:
chore: adjust keys

sethkfman committed:
Merge branch 'main' into feat/add-notifications-polishing

tommasini commented:
@Jonathansoufer I can see that this PR have multiple E2E failures, let's address those first

Cal-L commented:
Let's fix the CI checks

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Commit:
chore: add basicFunctionality dependency on disabling

Commit:
chore: bump notifications controllers

Commit:
chore: add back metametrics Id generator on Engine

Commit:
chore: add profile sync enablement for new users

Commit:
chore: add profile sync enablement to import wallet flow

Commit:
bump controllers version

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Commit:
bump notification controllers version

Commit:
fix lint

Commit:
chore: add mission actions

Commit:
chore: update snapshot

Commit:
chore: add missing events

Commit:
chore: add missing actions/events

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

tommasini commented on app/components/Views/Settings/NotificationsSettings/NotificationOptionToggle/index.tsx:
```diff
@@ -12,65 +12,96 @@ import {
+      {type === NotificationsToggleTypes.ACTIONS && icon ? (
         <Icon
-          name={icon}
+          name={icon as IconName}
```
Instead of casting here as IconName we could type it [here](https://github.com/MetaMask/metamask-mobile/blob/72339d1045c4a2ad4c107710ec0be73991ee1e03/app/components/Views/Settings/NotificationsSettings/NotificationOptionToggle/index.tsx#L25) wdyt?

Jonathansoufer replied:
This was based on another suggestions coming from a review which led to `icon?: AvatarAccountType | IconName;`. when we avoid the usage of `any`.

tommasini commented on app/components/Views/Settings/NotificationsSettings/NotificationOptionToggle/index.tsx:
```diff
@@ -12,65 +12,96 @@ import {
           variant={AvatarVariant.Account}
-          type={icon}
-          accountAddress={description}
+          type={icon as AvatarAccountType}
```
Hmm I see why my comment above could not make sense

Jonathansoufer replied:
yep. answered above already.

tommasini commented:
When opening the network bottom sheed on the notification is intended to the button of the "View on block explorer" keep being visible? 
Recording time: 0.50 second
<img width="504" alt="image" src="https://github.com/user-attachments/assets/b05c5941-15f2-4e85-ae33-25f05893cb4e">

tommasini commented:
Curious question: Why the swap doesn't have the button to see at the "View Block Explorer" ?
Recording time: 1:27 min
<img width="546" alt="image" src="https://github.com/user-attachments/assets/c6c84d7f-2a65-42d5-b456-5ce8908dafec">

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Jonathansoufer commented:
> Curious question: Why the swap doesn't have the button to see at the "View Block Explorer" ? Recording time: 1:27 min <img alt="image" width="546" src="https://private-user-images.githubusercontent.com/46944231/357609742-c6c84d7f-2a65-42d5-b456-5ce8908dafec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2MTgxNDUsIm5iZiI6MTcyMzYxNzg0NSwicGF0aCI6Ii80Njk0NDIzMS8zNTc2MDk3NDItYzZjODRkN2YtMmE2NS00MmQ1LWI0NTYtNWNlODkwOGRhZmVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDA2NDQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1YjQzNGM3OGUzMzk3YWNkZWE2Yzc5MGQ4N2Y4ZGVmMDdjMmFkYTM4OGExZGNhMTE5YjM0OGY2MTllNzY5ZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZPftZ3J9zy8p6QUF32wUa77uNUymN3KV1DJiJ3FBKEw">

Good catch. Fixed.

Commit:
add missing explorer btn

tommasini committed:
fix: useBlockExplorer hook (#10613)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

tommasini commented on app/components/Views/Settings/NotificationsSettings/index.tsx:
```diff
@@ -139,41 +136,49 @@ const NotificationsSettings = ({ navigation, route }: Props) => {
     </>
   );
 
+  const renderAccounts = useCallback(() => {
+    const refetchAccountSettings = async () => {
+      await accountSettingsProps.update(accountAddresses);
+    };
+
+    return accounts.map((account) => (
+      <NotificationOptionToggle
+        type={NotificationsToggleTypes.ACCOUNT}
+        icon={accountAvatarType}
+        key={account.address}
+        title={account.name}
+        address={account.address}
+        disabledSwitch={accountSettingsProps.initialLoading}
+        isLoading={accountSettingsProps.accountsBeingUpdated.includes(
+          account.address,
+        )}
+        isEnabled={
+          accountSettingsProps.data?.[account.address.toLowerCase()] ?? false
+        }
+        refetchAccountSettings={refetchAccountSettings}
+      />
+    ));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountSettingsProps]);
```
I wonder for we not need to use the  `// eslint-disable-next-line react-hooks/exhaustive-deps` we could refactor to refetchAccountSettings outside, since we do not want this to re render when the accounts coming from `useAccounts` change, we do not need it to add it to the dependency that way. 

```suggestion
    const refetchAccountSettings = async () => {
    await accountSettingsProps.update(accountAddresses);
  };

  const renderAccounts = useCallback(() => {
    return accounts.map((account) => (
      <NotificationOptionToggle
        type={NotificationsToggleTypes.ACCOUNT}
        icon={accountAvatarType}
        key={account.address}
        title={account.name}
        address={account.address}
        disabledSwitch={accountSettingsProps.initialLoading}
        isLoading={accountSettingsProps.accountsBeingUpdated.includes(
          account.address,
        )}
        isEnabled={
          accountSettingsProps.data?.[account.address.toLowerCase()] ?? false
        }
        refetchAccountSettings={refetchAccountSettings}
      />
    ));
  }, [
    accounts,
    accountAvatarType,
    accountSettingsProps.initialLoading,
    accountSettingsProps.accountsBeingUpdated,
    accountSettingsProps.data,
    refetchAccountSettings,
  ]);
```

I'm assuming that this was done to avoid a infinite loop, but I do not see why that would cause it, is the `useAccounts` hook always returning a different accounts object? If yes, more likely we should create an issue to try to see where the problem lives, wdyt?

tommasini reviewed (changes requested):
The PR LGTM! Just left a comment that we would want to address!

Jonathansoufer committed:
Update app/components/Views/Settings/NotificationsSettings/index.tsx

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>

Jonathansoufer committed:
Merge branch 'main' into feat/add-notifications-polishing

Commit:
fix lint problem
